test/test_autograd.py

```diff
@@ -4499,6 +4502,105 @@ def fn(*inputs):
 
             setattr(TestAutogradDeviceType, test_name, do_test)
 
+class TestAutogradComplex(TestCase):
```

Comment:
Should this class be made device generic so it runs on CPU and CUDA?

Reply:
autograd tests run both cpu and cuda already

Reply:
These tests won't run on both but that's OK for now.